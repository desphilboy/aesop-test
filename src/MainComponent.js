import React, { useState } from 'react';
import { ExpansionPanel, Button } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
import Typography from '@material-ui/core/Typography';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import Drawer from '@material-ui/core/Drawer';

const ItemDisplay = ({ item }) => (
	<div
		style={{
			padding: '10px',
		}}>
		<div> {item.name} </div>
		{item &&
			item.variants &&
			item.variants.map(variant => (
				<div
					style={{
						padding: '10px',
						minWidth: '360px',
						marginBottom: '40px',
						marginTop: '20px',
						border: 'solid 1px pink',
						borderRadius: '10px',
					}}>
					<div> {variant.sku} </div>
					<div>In Stock: {`${variant.inStock}`}</div>
					<img
						src={`https://www.aesop.com/au${variant.thumbnail}`}
						style={{ width: '240px', height: '240px' }}
					/>
					<div>{variant.price}</div>
				</div>
			))}
	</div>
);

const ItemComponent = ({ item }) => {
	const [toggle, setToggle] = useState(false);

	return (
		<div>
			<Button
				onClick={() => {
					item.drawerIsOpen = true;
					setToggle(!toggle);
				}}>
				{item.name}
			</Button>
			<Drawer
				anchor="left"
				open={!!item.drawerIsOpen}
				onClose={() => {
					item.drawerIsOpen = false;
					setToggle(!toggle);
				}}>
				<ItemDisplay item={item} />
			</Drawer>
		</div>
	);
};

const useStyles = makeStyles(theme => ({
	root: {
		width: '100%',
	},
	heading: {
		fontSize: theme.typography.pxToRem(15),
		fontWeight: theme.typography.fontWeightRegular,
	},
}));

export const MainComponent = ({ data }) => {
	const classes = useStyles;

	return (
		<div className={classes.root}>
			{data &&
				data.categories &&
				data.categories.map(category => (
					<ExpansionPanel key={category.name}>
						<ExpansionPanelSummary
							expandIcon={<ExpandMoreIcon />}
							aria-controls="panel1a-content"
							id="panel1a-header">
							<Typography className={classes.heading}>{category.name}</Typography>
						</ExpansionPanelSummary>
						<div style={{ paddingLeft: '30px' }}>
							{category.items.map(subCat => (
								<ExpansionPanel key={subCat.name}>
									<ExpansionPanelSummary
										key={subCat.name}
										expandIcon={<ExpandMoreIcon />}
										aria-controls="panel1a-content"
										id="panel1a-header">
										<Typography className={classes.heading}>
											{subCat.name}
										</Typography>
									</ExpansionPanelSummary>
									<div style={{ paddingLeft: '60px' }}>
										{subCat.items.map(item => (
											<ItemComponent key={item.name} item={item} />
										))}
									</div>
								</ExpansionPanel>
							))}
						</div>
					</ExpansionPanel>
				))}
		</div>
	);
};
