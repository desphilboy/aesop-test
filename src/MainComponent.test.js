import React from 'react';
import { render } from '@testing-library/react';
import { ItemDisplay } from './MainComponent.js';

describe('Components', () => {
	describe('ItemDisplay', () => {
		it('can render with invalid data', () => {
			const wrapper = render(<ItemDisplay item={{}} />);

			expect(wrapper).toMatchSnapshot();
		});

		it('renders with data', () => {
			const wrapper = render(
				<ItemDisplay
					item={{
						name: 'some product name',
						variants: [
							{
								sku: 'BALOO',
								inStock: false,
								thumbnail: '/url/moorl/kaloo.jpg',
								price: '80$',
							},
							{
								sku: 'PALOO',
								inStock: true,
								thumbnail: '/url/moorl/Paloo.jpg',
								price: '90$',
							},
						],
					}}
				/>,
			);

			expect(wrapper).toMatchSnapshot();
		});

		it('shows all variants', async () => {
			const { findAllByTestId } = render(
				<ItemDisplay
					item={{
						name: 'some product name',
						variants: [
							{ sku: 'BALOO', inStock: false, thumbnail: '/url/moorl/kaloo.jpg' },
							{ sku: 'PALOO', inStock: true, thumbnail: '/url/moorl/Paloo.jpg' },
							{ sku: 'TALOO', inStock: true, thumbnail: '/url/moorl/taloo.jpg' },
						],
					}}
				/>,
			);

			expect(await findAllByTestId('variant-show')).toHaveLength(3);
		});
	});
});
