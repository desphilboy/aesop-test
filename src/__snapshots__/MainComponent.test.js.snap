// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components ItemDisplay can render with invalid data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="padding: 10px;"
      >
        <div>
           
           
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="padding: 10px;"
    >
      <div>
         
         
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Components ItemDisplay renders with data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="padding: 10px;"
      >
        <div>
           
          some product name
           
        </div>
        <div
          data-testid="variant-show"
          style="padding: 10px; min-width: 360px; margin-bottom: 40px; margin-top: 20px; border: 1px solid pink; border-radius: 10px;"
        >
          <div>
             
            BALOO
             
          </div>
          <div>
            In Stock: 
            false
          </div>
          <img
            src="https://www.aesop.com/au/url/moorl/kaloo.jpg"
            style="width: 240px; height: 240px;"
          />
          <div>
            80$
          </div>
        </div>
        <div
          data-testid="variant-show"
          style="padding: 10px; min-width: 360px; margin-bottom: 40px; margin-top: 20px; border: 1px solid pink; border-radius: 10px;"
        >
          <div>
             
            PALOO
             
          </div>
          <div>
            In Stock: 
            true
          </div>
          <img
            src="https://www.aesop.com/au/url/moorl/Paloo.jpg"
            style="width: 240px; height: 240px;"
          />
          <div>
            90$
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="padding: 10px;"
    >
      <div>
         
        some product name
         
      </div>
      <div
        data-testid="variant-show"
        style="padding: 10px; min-width: 360px; margin-bottom: 40px; margin-top: 20px; border: 1px solid pink; border-radius: 10px;"
      >
        <div>
           
          BALOO
           
        </div>
        <div>
          In Stock: 
          false
        </div>
        <img
          src="https://www.aesop.com/au/url/moorl/kaloo.jpg"
          style="width: 240px; height: 240px;"
        />
        <div>
          80$
        </div>
      </div>
      <div
        data-testid="variant-show"
        style="padding: 10px; min-width: 360px; margin-bottom: 40px; margin-top: 20px; border: 1px solid pink; border-radius: 10px;"
      >
        <div>
           
          PALOO
           
        </div>
        <div>
          In Stock: 
          true
        </div>
        <img
          src="https://www.aesop.com/au/url/moorl/Paloo.jpg"
          style="width: 240px; height: 240px;"
        />
        <div>
          90$
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
